lib/chainweb/client/default.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Kadena.Chainweb.Client.Default do
+  @moduledoc """
+  Hackney HTTP client implementation.
+
+  This implementation allows you to use your own JSON encoding library. The default is Jason.
+  """
+
+  @behaviour Kadena.Chainweb.Client.Spec
+
+  @type status :: pos_integer()
+  @type headers :: [{binary(), binary()}, ...]
+  @type body :: binary()
+  @type success_response :: {:ok, status(), headers(), body()}
+  @type error_response :: {:error, status(), headers(), body()} | {:error, any()}
+  @type client_response :: success_response() | error_response()
+  @type parsed_response :: {:ok, map()} | {:error, Keyword.t()}
+
+  @impl true
+  def request(path, method, headers \\ [], body \\ "", opts \\ []) do
+    options = http_options(opts)
+
+    method
+    |> http_client().request(path, headers, body, options)
+    |> handle_response()
+  end
+
+  @spec handle_response(response :: client_response()) :: parsed_response()
+  defp handle_response({:ok, status, _headers, body}) when status in 200..299 do
+    decoded_body = json_library().decode!(body, keys: :atoms)
+    {:ok, decoded_body}
+  end
+
+  defp handle_response({:ok, status, _headers, body}) when status in 400..599,
+    do: {:error, [chainweb: body]}
```

Comment:
@santiagoboterokommit, It's not necessary to add this line here `decoded_body = json_library().decode!(body, keys: :atoms)`